ts/private/ts_project.bzl

```diff
@@ -85,21 +85,21 @@ See https://github.com/aspect-build/rules_ts/issues/228 for more details.
     # Add user specified arguments *before* rule supplied arguments
     arguments.add_all(ctx.attr.args)
 
-    outdir = _lib.join(ctx.label.package, ctx.attr.out_dir) if ctx.attr.out_dir else ctx.label.package
-    if outdir == "":
-        outdir = "."
+    outdir = _lib.join(
+        ctx.label.workspace_root,
+        _lib.join(ctx.label.package, ctx.attr.out_dir) if ctx.attr.out_dir else ctx.label.package,
+    )
+    tsconfig_path = ctx.file.tsconfig.path.removeprefix("%s/" % ctx.bin_dir.path)
     arguments.add_all([
         "--project",
-        ctx.file.tsconfig.short_path,
+        tsconfig_path,
         "--outDir",
         outdir,
         "--rootDir",
         _lib.calculate_root_dir(ctx),
     ])
     if len(typings_outs) > 0:
-        declaration_dir = _lib.join(ctx.label.package, typings_out_dir)
-        if declaration_dir == "":
-            declaration_dir = "."
+        declaration_dir = _lib.join(ctx.label.workspace_root, ctx.label.package, typings_out_dir)
```

Comment:
I think we still need
```
        if declaration_dir == "":
            declaration_dir = "."
```
since `ctx.label.workspace_root` will be "" for the `__main__` workspace

Reply:
We should also add a test case for this one. The case is a `ts_project` in the root BUILD file of a repository with `declaration = True` (may as well make it in an `e2e` so we're not polluting the root package of `rules_ts).

Reply:
Actually this case passes correctly as is. `_lib.join()` defaults to "." if the result is empty, so I don't think we need to change anything.

https://github.com/aspect-build/rules_ts/blob/ce5df5155c96cb84826176462a310b81a62f8f10/ts/private/ts_lib.bzl#L142-L146

I updated the existing example to use `declaration = True` and added a `build_test()` directly to verify that it builds in both workspaces. I think that should at least serve as a regression test going forward.

Reply:
Oh nice. That works 👍